common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationSupport.java

```diff
@@ -157,18 +164,6 @@ public static boolean execICU(final UTF8String l, final UTF8String r,
 
   private static class CollationAwareUTF8String {
```

Comment:
Do we still need that? Looks like dead code

Reply:
there will be some other methods here (some currently open PRs use `CollationAwareUTF8String`), but we will now remember to challenge more rigorously all collation aware UTF8String implementations (like you did for `matchAt`)